predictive_test_selection/collect_test_failure_data.py

```diff
@@ -0,0 +1,168 @@
+import logging
+import os
+import subprocess
+import time
+import xml.etree.ElementTree as ET
+from datetime import datetime
+from enum import IntFlag, auto
+from typing import Any, List, Mapping, Sequence, Tuple
+
+import requests
+
+LOGGER = logging.getLogger(__name__)
+
+
+class CommonFileExtension(IntFlag):
+    other = auto()
+    mo = auto()
+    po = auto()
+    pyc = auto()
+    json = auto()
+    dat = auto()
+    pyi = auto()
+    py = auto()
+
+
+class DataParser:
+    """
+        This class fetches all information regarding a specific code change and the tests that are run on this specific commit
+        and sends it all to the influxdb database.
+
+        Information regarding the code change:
+            x commit_hash
+            x n_changes_3  \
+            x n_changes_14  -> Changes made to the files involved in this change in the last 3, 14 and 56 days
+            x n_changes_56 /
+            x file_cardinality -> Number of files involved in this change
+            x file_extension -> File extensions of the files involved in this change
+
+        Information regarding the tests:
+            x test_fqn -> fully qualified name of the test case: <test_file_name>.<test_case>
+            x test_failed -> integer denoting whether this test failed or not.
+    """
+
+    def __init__(self):
+        self.code_change_data: Mapping[str, Any] = {}
+        self.test_result_data: Mapping[str, int] = {}
+
+    def parse_code_change(self) -> None:
+        """
+        Collect all data pertaining to the code change in the currently checked out branch:
+        - the commit hash
+        - the number of modified files
+        - the file extensions of modified files
+        - the number of changes to the modified files in the past 3, 14 and 56 days
+        """
+        data = {
+            "commit_hash": None,
+            "file_cardinality": None,
+            "file_extension": None,
+            "n_changes": None,
+        }
+
+        # Get latest commit hash:
+        data["commit_hash"] = subprocess.check_output(["git", "log", "--pretty=%H", "-1"]).strip()
+
+        # Get current branch
+        cmd = ["git", "rev-parse", "--abbrev-ref", "HEAD"]
+        current_branch = subprocess.check_output(cmd).strip().decode()
+
+        cmd = ["git", "diff", f"master...{current_branch}", "--name-only"]
```

Comment:
This code doesn't take into account the situation where the diff is empty. In that case we don't want to collect any data, I guess.

Reply:
What could cause this situation to arise ? And indeed, I don't think we want to collect any data in this case.